tasks/configure-allowlist.ts

```diff
@@ -0,0 +1,43 @@
+import '@nomiclabs/hardhat-ethers'
+import { parse } from 'csv-parse'
+import { task } from 'hardhat/config'
+import fs from 'fs'
+
+interface AllowlistRow {
+  address: string
+  amount: number
+}
+
+task('configure-allowlist', 'Configure the Allowlist')
+  .addParam('csvpath')
+  .addParam('contractaddress')
+  .setAction(async ({ csvpath, contractaddress }, hre) => {
+    return new Promise<void>((resolve, reject) => {
```

Comment:
can we try to use `await` and `async` ? i don't know if something like that would work? https://csv.js.org/parse/api/async_iterator/

it's always nicer to use the await / async syntax

Reply:
Thanks, thats better indeed